.github/workflows/compound-java-dev.yml

```diff
@@ -0,0 +1,122 @@
+name: Compound Dev Java build and Docker/Sonatype publish
+
+on:
+  workflow_call:
+    inputs:
+      docker-username:
+        required: false
+        type: string
+      build-target:
+        required: true
+        type: string
+        description: 'Use values Docker or Sonatype or both'
+      runsOn:
+        required: false
+        type: string
+        default: 'ubuntu-20.04'
+# Trivy block
+      scanner-enabled:
+        required: false
+        type: boolean
+        default: true
+        description: 'Whether scanning for vulnerabilities is enabled'
+      trivy-target:
+        required: false
+        type: string
+        default: 'os,library'
+        description: 'What Trivy will scan: image, library or both'
+      trivy-sensitivity:
+        required: false
+        type: string
+        default: 'CRITICAL,HIGH,MEDIUM'
+        description: 'Adjusting the Trivy Scanner Sensitivity'
+
+    outputs:
+      version:
+        description: "Generated version of the image"
+        value: ${{ jobs.build.outputs.version }}
+    secrets:
+      docker-password:
+        required: false
+      sonatypeUsername:
+        required: false
+      sonatypePassword:
+        required: false
+      sonatypeSigningKey:
+        required: false
+      sonatypeSigningPassword:
+        required: false
+
+jobs:
+  prebuild-job:
+    name: Prebuild Job
+    uses: th2-net/.github/.github/workflows/compound-prebuild-java-dev-workflow.yml@main
+    with:
+      runsOn: ${{ inputs.runsOn }}
+
+  # if Docker
+  docker-push:
+    name: Build Docker image and push to ghcr.io
+    if: contains(inputs.build-target, 'Docker') || contains(inputs.build-target, 'docker')
+    needs: [prebuild-job]
+    runs-on: ${{ inputs.runsOn }}
+    steps:
+      - name: Show custom release version
+        run: echo ${{ needs.prebuild-job.outputs.version }}
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ inputs.docker-username }}
+          password: ${{ secrets.docker-password }}
+      - name: Get repository name
+        run: echo "REPOSITORY_NAME=$(echo '${{ github.repository }}' | awk -F '/' '{print $2}')" >> $GITHUB_OUTPUT
+        id: meta
+      - name: Build and push
+        id: docker_build
+        uses: docker/build-push-action@v3
+        with:
+          push: true
+          tags: ghcr.io/${{ github.repository }}:${{ needs.prebuild-job.outputs.version }}
+          labels: com.exactpro.th2.${{ steps.meta.outputs.REPOSITORY_NAME }}=${{ needs.prebuild-job.outputs.versionNumber }}
```

Comment:
Could you please extract to separate first level template (the simplest operation)